src/Sound/Tidal/Params.hs

```diff
@@ -179,6 +179,18 @@ end = pF "end"
 gain :: Pattern Double -> ControlPattern
 gain = pF "gain"
 
+-- | 'vol' is a version of gain with a built-in limiting that clips the provided
+-- values to stay between 0 and 1.999 in order to protect against common
+-- typing errors causing dangerously or uncomfortably loud sounds.
+vol :: Pattern Double -> ControlPattern
+vol = gain . clip 0 1.999
+
+-- | clip constrains a fractional pattern to stay within the limits
+-- specified by lowerBound and upperBound.
+clip :: Ord a => Pattern a -> Pattern a -> Pattern a -> Pattern a
+clip lowerBound upperBound x = f <$> lowerBound <*> upperBound <*> x
```

Comment:
I think it's worth making a version `_clip` for internal use (such as above) that doesn't have the overhead of patterned parameters. Then `clip` can be defined relative to that.
(Also I think it's best to use `*>` rather than `<*>` in this kind of situation, so that the structure of the final parameter takes precedence)

Reply:
Not sure when I'll have time for this, so please feel free to just do what you suggest.